Source/Dafny/Options/TargetOption.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Dafny.Plugins;
+
+namespace Microsoft.Dafny;
+
+public class TargetOption : CommandLineOption<Compiler> {
+  public static readonly TargetOption Instance = new();
+  public override object GetDefaultValue(DafnyOptions options) =>
+    options.Plugins.SelectMany(p => p.GetCompilers()).First(c => c.TargetId == "cs");
+
+  public override string LongName => "target";
+  public override string ShortName => "t";
+  public override string Category => "Compilation options";
+
+  public override string Description => @"
+cs (default) - Compilation to .NET via C#
+go - Compilation to Go
+js - Compilation to JavaScript
+java - Compilation to Java
+py - Compilation to Python
+cpp - Compilation to C++
```

Comment:
Should this string be dynamically generated based on available compiler plugins?